src/NetscriptFunctions.ts

```diff
@@ -98,6 +98,11 @@ interface NS extends INS {
   stanek: IStanek;
 }
 
+//interface guard
+function isNS(object: any): boolean {
+  return (object as NS).gang !== undefined;
```

Comment:
Can you explain why this is needed? `typeof data` when data is an NS object should return neither "string" nor "number" (presumably it returns "NS") - so the existing checks look like they would suffice.

Reply:
hmmmmmm ill double check... but the ticket implied that you could pass through the whole ns object, is it because string and number is a object lower down under the hood maybe?... (not sure if js works this way)

its possible this one is no longer an issue and i just derped and never tried it before the fix 😆 .

But the custom check was the recomended way online to enforce that something is not a certain type of interface.